### PR TITLE
#167377556 Users should be able to share articles on social platforms

### PR DESCRIPTION
**Pull request by Bastieno**
#### What does this PR do?
This pull request enables a user to share an article on Facebook, Twitter or Linkedin

#### Description of Task to be completed
- implement share article functionality for articles
- only logged in users can share an article
- social platform supported are Facebook, Twitter, and Linkedin

#### How should this be manually tested?
- clone this branch
- run `npm install`
- using postman send a `POST` request to `/api/v1/articles/:slug/share/{social_platform}`

#### Any Background Context You want to provide?
N/A

#### What are the relevant pivotal tracker stories?

[#167377556](https://pivotaltracker.com/story/show/167377556)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

**Comment by ibkadeeko** [comment]
Please change the PR title to use the team conventions and your commit message should have `Finishes` not `Delivers`

**Review by ibkadeeko — CHANGES_REQUESTED** [review]
Please change the PR title to use the team conventions and your commit message should have `Finishes` not `Delivers`

**Comment by Bastieno** [comment]
> Please change the PR title to use the team conventions and your commit message should have `Finishes` not `Delivers`

Done!

File: tests/routes/articles.test.js [hunk 1]
```diff
@@ -604,3 +605,96 @@ describe('GET /articles/:articleId/comments', () => {
     response.body.error.message.should.eql('Non-existing articleId');
   });
 });
+
+describe('GET /articles/:articleId/share/{socialPlatform}', () => {
```
Comment (Bastieno) [inline tests/routes/articles.test.js:609]: Change to article slug

File: src/helpers/article.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
+import dotenv from 'dotenv';
 import Models from '../models/index';
 import { ApplicationError } from './errors';
 
+dotenv.config();
+
+const appUrl = process.env.APP_URL;
 const uuidRegularExpression = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/;
```
Comment (Bastieno) [inline src/helpers/article.js:8]: Change this

File: src/routes/article.js [hunk 3]
```diff
@@ -373,4 +374,58 @@ router.delete(
...
+ *       - application/json
+ *     parameters:
+ *        - in: path
+ *          name: slug
+ *          required: true
+ *     responses:
+ *       200:
+ *         description: success
+ */
+router.get('/:slug/share/facebook', asyncWrapper(verifyToken), asyncWrapper(shareArticle));
```
Comment (Bastieno) [inline src/routes/article.js:393]: Merge all three

File: .env.example [hunk 4]
```diff
@@ -16,3 +16,4 @@ SENDGRID_ACTIVATION_TEMPLATE=
 CLOUDINARY_CLOUD_NAME=
 CLOUDINARY_API_KEY=
 CLOUDINARY_API_SECRET=
+APP_URL=
```
Comment (Bastieno) [inline .env.example:19]: Get this from express request

File: src/controllers/article.js [hunk 5]
```diff
@@ -367,5 +372,34 @@ export default {
...
+   * @return {Object} - the response from the server
+   */
+  shareArticle: async (request, response) => {
+    const { slug } = request.params;
+    const article = await Articles.findOne({ where: { slug } });
+
+    if (!article) throw new ApplicationError(404, 'Invalid slug');
+
+    if (request.url.search(/\/facebook/) > 0) {
+      const facebookShareLink = createFacebookShareLink(slug);
```
Comment (Bastieno) [inline src/controllers/article.js:391]: Pass in the url and slug to make just one function

**Pull request closed without merging**